project.go

```diff
@@ -0,0 +1,392 @@
+		c.fs.Var(&c.envs, "envs", "comma-separated list of environments to use (defaults to dev,qa,stage,production)")
+		c.fs.Var(&c.envs, "e", "comma-separated list of environments to use (defaults to env,qa,stage,production)")
+		c.envs.Set("dev,qa,stage,production")
+	}
+	return c.fs
+}
+
+func (c *projectCreate) createApps(envs []Environment, client *cmd.Client) ([]map[string]string, error) {
+	createdApps := make([]map[string]string, 0, len(envs))
+	appNames := make([]string, 0, len(envs))
+	for _, env := range envs {
+		appName := fmt.Sprintf("%s-%s", c.name, env.Name)
+		app, err := createApp(client, createAppOptions{
+			name:     appName,
+			plan:     c.plan,
+			platform: c.platform,
+			pool:     env.poolName(),
+			team:     c.team,
+		})
+		if err != nil {
+			deleteApps(appNames, client)
+			return nil, fmt.Errorf("failed to create the project in env %q: %s", env.Name, err)
+		}
+		app["name"] = appName
+		app["dnsSuffix"] = env.DNSSuffix
+		createdApps = append(createdApps, app)
+		appNames = append(appNames, appName)
+	}
+	return createdApps, nil
+}
+
+func (c *projectCreate) setCNames(apps []map[string]string, client *cmd.Client) error {
+	for _, app := range apps {
+		reqURL, err := cmd.GetURL(fmt.Sprintf("/apps/%s/cname", app["name"]))
+		if err != nil {
+			return err
+		}
+		cname := fmt.Sprintf("%s.%s", c.name, app["dnsSuffix"])
+		v := make(url.Values)
+		v.Set("cname", cname)
+		req, err := http.NewRequest("POST", reqURL, strings.NewReader(v.Encode()))
+		if err != nil {
+			return err
+		}
+		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		resp.Body.Close()
+	}
+	return nil
+}
+
+func (c *projectCreate) filterEnvironments(envs []Environment, names []string) []Environment {
+	var filtered []Environment
+	for _, e := range envs {
+		for _, name := range names {
+			if e.Name == name {
+				filtered = append(filtered, e)
+				break
+			}
+		}
+	}
+	return filtered
+}
+
+type commaSeparatedFlag struct {
+	values []string
+}
+
+func (f *commaSeparatedFlag) Values() []string {
+	return f.values
+}
+
+func (f *commaSeparatedFlag) String() string {
+	return strings.Join(f.values, ",")
+}
+
+func (f *commaSeparatedFlag) Set(v string) error {
+	f.values = strings.Split(v, ",")
+	return nil
+}
+
+func (f *commaSeparatedFlag) validate(validValues []string) error {
+	var invalidValues []string
+	for _, cv := range f.values {
+		var found bool
+		for _, validValue := range validValues {
+			if validValue == cv {
+				found = true
+				break
+			}
+		}
+		if !found {
+			invalidValues = append(invalidValues, cv)
+		}
+	}
+	if len(invalidValues) > 0 {
+		return fmt.Errorf("invalid values: %s (valid options are: %s)", strings.Join(invalidValues, ", "), strings.Join(validValues, ", "))
+	}
+	return nil
+}
+
+type projectRemove struct {
+	cmd.ConfirmationCommand
+	name string
+	fs   *gnuflag.FlagSet
+}
+
+func (c *projectRemove) Info() *cmd.Info {
```

Comment:
Since this is Open source and all shouldn't it be better to have all names and desc in a separate file? for editing/translating purpose?